sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -318,7 +318,8 @@ case class SampleExec(
         v => s"""
           | $v = new $samplerClass<UnsafeRow>($lowerBound, $upperBound, false);
           | $v.setSeed(${seed}L + partitionIndex);
-         """.stripMargin.trim)
+         """.stripMargin.trim,
+        forceInline = true)
```

Comment:
why do we need this?

Reply:
Generic arrays not allowed;
```
/* 019 */   private org.apache.spark.util.random.BernoulliCellSampler<UnsafeRow>[] sample_mutableStateArray_0 = new org.apache.spark.util.random.BernoulliCellSampler<UnsafeRow>[1];
--- 
 Cause: java.util.concurrent.ExecutionException: org.codehaus.commons.compiler.CompileException: failed to compile:
(Line 40, Column 101) generic array creation
```

Reply:
Though this may cause problems leading to the 64KB limit issue. So if we are not including the jdk support I'd be against this change...

Reply:
aha, ok. let's us wait for the other developer's comments.